homeassistant/components/mqtt/__init__.py

```diff
@@ -163,42 +162,45 @@ def embedded_broker_deprecated(value):
 
 CONFIG_SCHEMA = vol.Schema(
     {
-        DOMAIN: vol.Schema(
-            {
-                vol.Optional(CONF_CLIENT_ID): cv.string,
-                vol.Optional(CONF_KEEPALIVE, default=DEFAULT_KEEPALIVE): vol.All(
-                    vol.Coerce(int), vol.Range(min=15)
-                ),
-                vol.Optional(CONF_BROKER): cv.string,
-                vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-                vol.Optional(CONF_USERNAME): cv.string,
-                vol.Optional(CONF_PASSWORD): cv.string,
-                vol.Optional(CONF_CERTIFICATE): vol.Any("auto", cv.isfile),
-                vol.Inclusive(
-                    CONF_CLIENT_KEY, "client_key_auth", msg=CLIENT_KEY_AUTH_MSG
-                ): cv.isfile,
-                vol.Inclusive(
-                    CONF_CLIENT_CERT, "client_key_auth", msg=CLIENT_KEY_AUTH_MSG
-                ): cv.isfile,
-                vol.Optional(CONF_TLS_INSECURE): cv.boolean,
-                vol.Optional(CONF_TLS_VERSION, default=DEFAULT_TLS_PROTOCOL): vol.Any(
-                    "auto", "1.0", "1.1", "1.2"
-                ),
-                vol.Optional(CONF_PROTOCOL, default=DEFAULT_PROTOCOL): vol.All(
-                    cv.string, vol.In([PROTOCOL_31, PROTOCOL_311])
-                ),
-                vol.Optional(CONF_EMBEDDED): vol.All(
-                    server.HBMQTT_CONFIG_SCHEMA, embedded_broker_deprecated
-                ),
-                vol.Optional(CONF_WILL_MESSAGE): MQTT_WILL_BIRTH_SCHEMA,
-                vol.Optional(CONF_BIRTH_MESSAGE): MQTT_WILL_BIRTH_SCHEMA,
-                vol.Optional(CONF_DISCOVERY, default=DEFAULT_DISCOVERY): cv.boolean,
-                # discovery_prefix must be a valid publish topic because if no
-                # state topic is specified, it will be created with the given prefix.
-                vol.Optional(
-                    CONF_DISCOVERY_PREFIX, default=DEFAULT_DISCOVERY_PREFIX
-                ): valid_publish_topic,
-            }
+        DOMAIN: vol.All(
+            cv.deprecated(CONF_TLS_VERSION, invalidation_version="0.115"),
```

Comment:
This will land in 0.112, are 3 versions reasonable?

Reply:
Yes